doc/explanation/scopes.rst

```diff
@@ -0,0 +1,36 @@
+Dune projects and workspaces
+============================
+
+Whenever Dune builds anything, it does so at the level of a *Dune workspace*. A
+Dune workspace is a set of Dune projects. A typical workspace consists of a
+single Dune project.
+
+A Dune project is defined by the presence of a
+:doc:`/reference/dune-project/index` file. Each Dune project extends over the
+subtree rooted at the directory containing the
+:doc:`/reference/dune-project/index` file, excluding any nested Dune projects.
+
+The root of the current workspace is determined by the outermost
+:doc:`/reference/dune-project/index` file in an ancestor of the current
+directory or by the presence of a :doc:`/reference/dune-workspace/index` file
+(see :ref:`finding-root` and :ref:`forcing-root` for details).
```

Comment:
```
Dune determines the root of the current workspace by finding the topmost ancestor directory containing a `dune-project` file or by the presence of a `dune-workspace` file.
```

An alternative wording, feel free to pick and choose. Q: Do we always pick the topmost dune-workspace file? What happens if we have A/B where A contains a dune-project and B a dune-workspace?